### PR TITLE
PHOENIX-5591 Fail indextool if split-merge happens during job run

### PR DESCRIPTION
**Pull request by gokceni**

### CODE REVIEW

File: phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/PhoenixIndexImportDirectReducer.java [hunk 1]
```diff
@@ -49,8 +50,42 @@
...
+
+    private static Integer regionCountForTesting = null;
+
+    public static void setRegionCountForTesting(Integer regionCount) {
+        regionCountForTesting = regionCount;
+    }
+
     @Override
     protected void cleanup(Context context) throws IOException, InterruptedException{
+        String strNumOfRegions = PhoenixConfigurationUtil.getNumOfRegions(context.getConfiguration());
```
Comment (priyankporwal) [inline phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/PhoenixIndexImportDirectReducer.java]: Isn't this the number of mappers? Can the reducer get it instead and compare with it?

File: phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/PhoenixIndexImportDirectReducer.java [hunk 2]
```diff
@@ -49,8 +50,42 @@
     private static final Logger LOGGER =
             LoggerFactory.getLogger(PhoenixIndexImportDirectReducer.class);
 
+    private static final String SPLIT_MERGE_ERROR_STR_FMT =
```
Comment (priyankporwal) [inline phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/PhoenixIndexImportDirectReducer.java]: This solution tackles Split-Only case. With concurrent merges, this is not a guarantee that region counts would differ.

File: phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/PhoenixIndexImportDirectReducer.java (hunk 1 above)
Comment (gokceni) [reply phoenix-core/src/main/java/org/apache/phoenix/mapreduce/index/PhoenixIndexImportDirectReducer.java]: It is not mappers. It is table region count. Not sure I understand what you are suggesting

**Comment by gokceni** [comment]
This Jira is not needed. It is fixed in IndexRegionObserver with sending LastCell to Hbase client.

**Pull request closed without merging**